libafl/src/stages/logics.rs

```diff
@@ -178,35 +209,54 @@ where
 
 impl<CB, E, EM, ST1, ST2, Z> Stage<E, EM, Z> for IfElseStage<CB, E, EM, ST1, ST2, Z>
 where
-    CB: FnMut(&mut Z, &mut E, &mut E::State, &mut EM, CorpusId) -> Result<bool, Error>,
+    CB: FnMut(&mut Z, &mut E, &mut E::State, &mut EM) -> Result<bool, Error>,
     E: UsesState,
     EM: UsesState<State = E::State>,
     ST1: StagesTuple<E, EM, E::State, Z>,
     ST2: StagesTuple<E, EM, E::State, Z>,
     Z: UsesState<State = E::State>,
+    E::State: HasNestedStageStatus,
 {
+    type Progress = NestedStageProgress;
+
     fn perform(
         &mut self,
         fuzzer: &mut Z,
         executor: &mut E,
         state: &mut E::State,
         manager: &mut EM,
-        corpus_idx: CorpusId,
     ) -> Result<(), Error> {
-        if (self.closure)(fuzzer, executor, state, manager, corpus_idx)? {
+        let current = state.current_stage()?;
```

Comment:
isn't current_stage()? always Some() when we are inside the stage? 

If we are resuming then we have the idx restored from previous stage, else we set it before entering perform() in StageTuple's implementation

Reply:
because we set here before entering any stage
https://github.com/AFLplusplus/LibAFL/pull/1780/files#diff-adaf83c80943a16f37541beaddf097804ee22246e6c7f6f0807d655d1ad65bf3R165

Reply:
ah I see. you increment the idx there.

Reply:
wait so that means when you are "normally (without resume)" executing the stage. the call to current_stage() always returns None().
I think the name current_stage() is not good. it's more like "last recorded checkpoint stage" right? (I can't think of a better name either)

Reply:
nevermind figure it out myself